egs/wsj/s5/steps/nnet3/get_egs.sh

```diff
@@ -183,14 +183,19 @@ if [ -f $dir/trans.scp ]; then
   train_subset_feats="$train_subset_feats transform-feats --utt2spk=ark:$data/utt2spk scp:$dir/trans.scp ark:- ark:- |"
 fi
 
+num_offsets=1
+if [ -f $data/offsets.scp ]; then
+  num_offsets=`feat-to-len --print-args=false scp:"head -n 1 $data/offsets.scp |"`
+fi
 if [ ! -z "$online_ivector_dir" ]; then
   ivector_dim=$(feat-to-dim scp:$online_ivector_dir/ivector_online.scp -) || exit 1;
+  ivector_dim=$[$ivector_dim/$num_offsets]
   echo $ivector_dim > $dir/info/ivector_dim
   ivector_period=$(cat $online_ivector_dir/ivector_period) || exit 1;
 
   ivector_opt="--ivectors='ark,s,cs:utils/filter_scp.pl $sdata/JOB/utt2spk $online_ivector_dir/ivector_online.scp | subsample-feats --n=-$ivector_period scp:- ark:- |'"
-  valid_ivector_opt="--ivectors='ark,s,cs:utils/filter_scp.pl $dir/valid_uttlist $online_ivector_dir/ivector_online.scp | subsample-feats --n=-$ivector_period scp:- ark:- |'"
-  train_subset_ivector_opt="--ivectors='ark,s,cs:utils/filter_scp.pl $dir/train_subset_uttlist $online_ivector_dir/ivector_online.scp | subsample-feats --n=-$ivector_period scp:- ark:- |'"
+  valid_ivector_opt="--ivectors='ark,s,cs:utils/filter_scp.pl $dir/valid_uttlist $online_ivector_dir/ivector_online.scp | subsample-feats --n=-$ivector_period scp:- ark:- | select-feats 0-$[$ivector_dim-1] ark:- ark:- |'"
```

Comment:
once you make the change I previously suggested about how the iVectors are stored, you won't need to make this change, you'll be changing the regular ivector-opt.  [note: this part of your changes are going to encounter merge problems with my PR here
https://github.com/kaldi-asr/kaldi/pull/1269/files
and you might want to consider doing your changes on top of that branch, but OK if you don't want to, I can resolve the issues later if not.]